packages/ui/src/checkbox-group/CheckboxGroup.test.tsx

```diff
@@ -0,0 +1,318 @@
+
+    // renders label with correct text
+    const label = getByTestId(getCheckboxGroupLabelTestId());
+    expect(label).toBeInTheDocument();
+    expect(label).toHaveTextContent(CHECKBOX_GROUP_LABEL);
+
+    // renders checkbox options with correct values
+    CHECKBOX_GROUP_OPTIONS.forEach(({ value }) => {
+      const option = checkboxGroup.querySelector(
+        `label[data-testid="${GET_TESTID_CHECKBOX(CHECKBOX_GROUP_NAME, value)}"]`,
+      );
+      if (!option) {
+        throw new Error(`Option ${value} not found`);
+      }
+      const input = option.querySelector("input[type='checkbox']");
+      expect(input).toBeInTheDocument();
+      expect(input).toHaveAttribute("value", value);
+      expect(input).toHaveAttribute("name", CHECKBOX_GROUP_NAME);
+      expect(input).not.toBeChecked();
+    });
+
+    // click on checkbox
+    const option1 = checkboxGroup.querySelector(
+      `label[data-testid="${GET_TESTID_CHECKBOX(CHECKBOX_GROUP_NAME, OPTION_1_VALUE)}"]`,
+    );
+    if (!option1) {
+      throw new Error("Option 1 not found");
+    }
+
+    await act(async () => {
+      await user.click(option1);
+    });
+
+    // check if correct checkbox is selected
+    const input1 = option1.querySelector("input[type='checkbox']");
+    expect(input1).toBeChecked();
+  });
+  // default value
+  it("renders item with default value as selected", () => {
+    const defaultValue = [OPTION_2_VALUE];
+    const { getByTestId } = render(
+      <CheckboxGroup
+        options={CHECKBOX_GROUP_OPTIONS}
+        label={CHECKBOX_GROUP_LABEL}
+        name={CHECKBOX_GROUP_NAME}
+        defaultValue={defaultValue}
+      />,
+    );
+
+    // renders checkbox group
+    const checkboxGroup = getByTestId(
+      GET_TESTID_CHECKBOX_GROUP(CHECKBOX_GROUP_NAME),
+    );
+    expect(checkboxGroup).toBeInTheDocument();
+
+    // get options and inputs and check if correct one is selected
+    const option1 = checkboxGroup.querySelector(
+      `label[data-testid="${GET_TESTID_CHECKBOX(CHECKBOX_GROUP_NAME, OPTION_1_VALUE)}"]`,
+    );
+    if (!option1) {
+      throw new Error("Option 1 not found");
+    }
+
+    const option2 = checkboxGroup.querySelector(
+      `label[data-testid="${GET_TESTID_CHECKBOX(CHECKBOX_GROUP_NAME, OPTION_2_VALUE)}"]`,
+    );
+    if (!option2) {
+      throw new Error("Option 2 not found");
+    }
+
+    const option3 = checkboxGroup.querySelector(
+      `label[data-testid="${GET_TESTID_CHECKBOX(CHECKBOX_GROUP_NAME, OPTION_3_VALUE)}"]`,
+    );
+    if (!option3) {
+      throw new Error("Option 3 not found");
+    }
+
+    const input1 = option1.querySelector("input[type='checkbox']");
+    const input2 = option2.querySelector("input[type='checkbox']");
+    const input3 = option3.querySelector("input[type='checkbox']");
+    expect(input1).not.toBeChecked();
+    expect(input2).toBeChecked();
+    expect(input3).not.toBeChecked();
+  });
+  // selected value overrides default value
```

Comment:
Can more than one value be selected in a checkbox group by default? I would expect that they could since checkbox groups are multi select.

Reply:
yes they can! the value/defaultValue are in string array forms. ie ["1"] or ["1","2"]. That comes out of the box with the react-aria library.